control/tap-ctl-ipc.c

```diff
@@ -187,6 +187,13 @@ tap_ctl_connect(const char *name, int *sfd)
 
 	memset(&saddr, 0, sizeof(saddr));
 	saddr.sun_family = AF_UNIX;
+
+	if (unlikely(strlen(name) > sizeof(saddr.sun_path) - 1)) {
```

Comment:
Or even pull the maximum allowed size out as a constant and use that, no parens required.